### PR TITLE
Added event arrival time differences to members

### LINKED ISSUE #43: Arrival time difference of consecutive trigger events
In order to calculate the total dead time of MAGIC observations, it is necessary to get the arrive time difference of consecutive events. I see that a MAGIC calibrated file has the time differences in the `MRawEvtHeader.fTimeDiff` branch, but currently it is not loaded by the `MAGICEventSource` module. And if we modify the module so as to load the time differences, the default `ArrayEventContainer` doesn't have a space to store them. 

Actually the time differences can be calculated by ourselves if we know the arrive times of events. As far as I understand, the time differences should be calculated by using both cosmic and pedestal events, but the current `MAGICEventSource` reads cosmic and pedestal events separately (depending on the option `use_pedestals`) and so I think we cannot reproduce the values of the `MRawEvtHeader.fTimeDiff` branch.

So there may be two solutions on this:

1. Create a custom timing container which can store a time difference value obtained from the `MRawEvtHeader.fTimeDiff` branch
2. Modify the `MAGICEventSource` to generate both cosmic and pedestal events simultaneously and calculate time differences in the process from calibrated to DL1

I think the solution 2 would be better since customizing the container should be avoided. But any comments or another ideas are very welcome.

### PR DESCRIPTION
**Pull request by YoshikiOhtani**
This pull request includes a new function to get event trigger time differences from the `MRawEvtHeader.fTimeDiff` branch, and they are stored in `self.event_time_diffs`. The time differences are computed considering all kind of events (cosmic, pedestal and calibration events), but here it extracts only those of cosmic events because the others are not used for dead time calculations. It closes #43.

### CODE REVIEW

**Review by aleberti — APPROVED** [review]

**Comment by YoshikiOhtani** [comment]
Thank you @aleberti for your approval. I would like to make a pull request to calculate the dead time using this new `event_time_diffs` member, so could you please again update the version of ctapipe_io_magic after merging this branch so that users does not need to install the master branch by themselves? Sorry for the convenience...!

**Comment by aleberti** [comment]
> Thank you @aleberti for your approval. I would like to make a pull request to calculate the dead time using this new `event_time_diffs` member, so could you please again update the version of ctapipe_io_magic after merging this branch so that users does not need to install the master branch by themselves? Sorry for the convenience...!

sure, I can do that. Let's wait for the review of Julian first, then I can merge and create the release. Then I will also update setup.py in MCP.

**Comment by jsitarek** [comment]
in principle looks fine, but I think this way you will have the time diffs disconnected from the actual events (as a separate array), this might be a problem if you want to cut a run with e.g. a zd cut and compute Teff from such a limited run. Better would be to keep the time difference with each event (there is no problem if some of them are later or removed by the cuts)

**Comment by YoshikiOhtani** [comment]
Hi @jsitarek, yes, my idea is also to keep the time diffs event-wise. I plan to modify the `magic_calib_to_dl1.py` script to add a new Field `time_diff` to the EventInfoContainer, and store event-wise values extracted from the array. In this way we will obtain a pandas DL1 data frame with a column time_diff, and then we can obtain time diffs of the events surviving any cuts.

**Comment by YoshikiOhtani** [comment]
By the way I pushed one more commit to add the unit (= sec) to the time_diff array just to be sure.

**Comment by YoshikiOhtani** [comment]
@aleberti I got an approval by Julian and so merged it to the master branch. It would be great if you could make a release of a new version, thanks!

**Comment by aleberti** [comment]
Done! https://pypi.org/project/ctapipe-io-magic/0.4.3/ and https://github.com/cta-observatory/ctapipe_io_magic/releases/tag/v0.4.3

**Pull request merged**
